### DISCUSSION TITLE
How to implement nested data based on polymorphic relationships?

### DISCUSSION THREAD

**Discussion by ahoiroman** [spatie/laravel-data#146] [Q&A] [answered]
Hello,

I am struggling with a little problem at the moment.

Given the scenario of a Model like `Video`, which morphs to either a `Comment` or a `Post`, how could I implement this relation in laravel-data?

At the moment, I am doing something like:

```
<?php

namespace App\Data;

use Spatie\LaravelData\{Data, Lazy};

class VideoData extends Data
{

    public function __construct(
        public Lazy|int  $id,
        public Lazy|Data $owner,
    ) {}

    public static function fromModel(Video $video): VideoData
    {
        return new self(
            id   : $video->id,
            owner: $video->targetable->getData()
        );
    }
}

```

As the `targetable` uses the Trait `WithData`, this is working as expected, but it does not quite good to use it this way. Any other ideas?

**Comment by rubenvanassche** [comment, marked as answer]
I don't see any problems in this case, though I would create a dedicated `OwnerData` object which can be created both from `Comme t` and `Post`. Then you could do something like `OwnerData::from($video->targetable)` which looks nice 😄
